CI: logs were not gathered for RuntimeConsulKVStore test

https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/2208/testReport/junit/(root)/runtime/RuntimeKVStoreTest_Consul_KVStore/

@eloycoto it seems the [1] function failed which prevented the logs from being gathered. 

[1]
```
	JustAfterEach(func() {
		vm.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
	})
```

This is in `test/runtime/kvstore.go`

The `JoinEP` log message was found in the logs because cilium was shutting down at that point

```
Jan 25 22:34:01 runtime systemd[1]: Stopping cilium...
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="Exiting due to signal" signal=terminated subsys=cilium-node-monitor
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="Stopped reading cilium agent events" subsys=cilium-node-monitor
Jan 25 22:34:01 runtime cilium-agent[9133]: level=error msg="Command execution failed" cmd="[llc -march=bpf -mcpu=probe -filetype=obj -o 3560_next/bpf_lxc.o]" error="signal: terminated" subsys=datapath-loader
Jan 25 22:34:01 runtime cilium-agent[9133]: level=error msg="Failed to compile bpf_lxc.o: signal: terminated" compiler-pid=9712 linker-pid=9713 subsys=datapath-loader
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="shutdown initiated" subsys=health-server
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="Stopped serving cilium health at http://[::]:4240" subsys=health-server
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="shutdown initiated" subsys=health-server
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="Stopped serving cilium health at http://[::]:4240" subsys=health-server
Jan 25 22:34:01 runtime cilium-agent[9133]: level=warning msg="JoinEP: Failed to compile" containerID= datapathPolicyRevision=0 debug=false desiredPolicyRevision=1 endpointID=3560 error="Failed to compile bpf_lxc.o: signal: terminated" identity=4 ipv4= ipv6= k8sPodName=/ params="&{Source:bpf_lxc.c Output:bpf_lxc.o OutputType:obj}" subsys=datapath-loader
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="Exiting due to signal" signal=terminated subsys=daemon
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="shutdown initiated" subsys=daemon
Jan 25 22:34:01 runtime cilium-agent[9133]: level=warning msg="Error while running monitor" error="Unable to read stdout from monitor: read |0: file already closed" subsys=monitor-launcher
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="Recompiled endpoint BPF program" BPFCompilationTime=459.165797ms containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=3560 error="Failed to compile bpf_lxc.o: signal: terminated" identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
Jan 25 22:34:01 runtime cilium-agent[9133]: level=error msg="Restoring endpoint state after BPF regeneration failed" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=3560 identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
Jan 25 22:34:01 runtime cilium-agent[9133]: level=error msg="Finished restoring endpoint state after BPF regeneration failed" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=3560 identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 [running]:
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 runtime/debug.Stack(0xbf0b0196318c310e, 0x2270c179b, 0x45e70e0)
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 github.com/cilium/cilium/pkg/lock.(*internalRWMutex).RUnlock(0xc001044b10)
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 	/home/vagrant/go/src/github.com/cilium/cilium/pkg/lock/lock_debug.go:84 +0xc3
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerateBPF(0xc0003f82c0, 0x2fc2cc0, 0xc000308fc0, 0xc00083cb40, 0x1, 0x1, 0x2f69940, 0xc000539ad0)
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 	/home/vagrant/go/src/github.com/cilium/cilium/pkg/endpoint/bpf.go:568 +0x85e
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerate(0xc0003f82c0, 0x2fc2cc0, 0xc000308fc0, 0xc00083cb40, 0x0, 0x0)
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 	/home/vagrant/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:309 +0x778
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 github.com/cilium/cilium/pkg/endpoint.(*Endpoint).Regenerate.func1(0xc0015c3f80, 0xc0001443f0, 0xc0003f82c0, 0x2fc2cc0, 0xc000308fc0)
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 	/home/vagrant/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:434 +0x230
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 created by github.com/cilium/cilium/pkg/endpoint.(*Endpoint).Regenerate
Jan 25 22:34:01 runtime cilium-agent[9133]: goroutine 664 	/home/vagrant/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:409 +0x89
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="[ERROR] close unix /var/run/cilium/cilium.sock->@: use of closed network connection" subsys=daemon
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="[ERROR] close unix /var/run/cilium/cilium.sock->@: use of closed network connection" subsys=daemon
Jan 25 22:34:01 runtime cilium-agent[9133]: level=warning msg="generating BPF for endpoint failed, keeping stale directory." containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=3560 file-path=3560_next_fail identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
Jan 25 22:34:01 runtime cilium-agent[9133]: level=info msg="Stopped serving cilium at unix:///var/run/cilium/cilium.sock" subsys=daemon
Jan 25 22:34:01 runtime cilium-agent[9133]: level=fatal msg="Error returned from non-returning Serve() call" error="<nil>" subsys=daemon
Jan 25 22:34:01 runtime systemd[1]: cilium.service: Main process exited, code=exited, status=1/FAILURE
Jan 25 22:34:01 runtime systemd[1]: cilium.service: Failed with result 'exit-code'.
Jan 25 22:34:01 runtime systemd[1]: Stopped cilium.
Jan 25 22:34:01 runtime systemd[1]: Started cilium.
```

Maybe the search for `JoinEP` needs to be adjusted and ignore messages that are after `Stopping cilium...`?

--- a/test/helpers/cilium.go
+++ b/test/helpers/cilium.go
@@ -656,12 +656,7 @@ func (s *SSHMeta) ValidateNoErrorsInLogs(duration time.Duration) {
 		}
 	}()
 
-	for _, message := range checkLogsMessages {
-		if strings.Contains(logs, message) {
-			fmt.Fprintf(CheckLogs, "⚠️  Found a %q in logs\n", message)
-			ginkgoext.Fail(fmt.Sprintf("Found a %q in Cilium Logs", message))
-		}
-	}
+	failIfContainsBadLogMsg(logs)
 
 	// Count part
 	for _, message := range countLogsMessages {

--- a/test/helpers/cons.go
+++ b/test/helpers/cons.go
@@ -236,7 +236,16 @@ const CiliumDefaultPreFlight = "cilium-pre-flight.yaml"
 // CiliumConfigMapPatch is the default Cilium ConfigMap patch to be used in all tests.
 const CiliumConfigMapPatch = "cilium-cm-patch.yaml"
 
-var checkLogsMessages = []string{panicMessage, deadLockHeader, segmentationFault, NACKreceived, RunInitFailed}
+// badLogMessages is a map which key is a part of a log message which indicates
+// a failure if the message does not contain any part from value list.
+var badLogMessages = map[string][]string{
+	panicMessage:      nil,
+	deadLockHeader:    nil,
+	segmentationFault: nil,
+	NACKreceived:      nil,
+	RunInitFailed:     {"signal: terminated", "signal: killed"},
+}
+
 var countLogsMessages = []string{contextDeadlineExceeded, ErrorLogs, WarningLogs, APIPanicked, selfishThresholdMsg}
 
 var ciliumCLICommands = map[string]string{

--- a/test/helpers/kubectl.go
+++ b/test/helpers/kubectl.go
@@ -1524,12 +1524,8 @@ func (kub *Kubectl) ValidateNoErrorsInLogs(duration time.Duration) {
 		}
 	}()
 
-	for _, message := range checkLogsMessages {
-		if strings.Contains(logs, message) {
-			fmt.Fprintf(CheckLogs, "⚠️  Found a '%s' in logs\n", message)
-			ginkgoext.Fail(fmt.Sprintf("Found a '%s' in Cilium Logs", message))
-		}
-	}
+	failIfContainsBadLogMsg(logs)
+
 	// Count part
 	for _, message := range countLogsMessages {
 		var prefix = ""

--- a/test/helpers/utils.go
+++ b/test/helpers/utils.go
@@ -393,3 +393,26 @@ func CanRunK8sVersion(ciliumVersion, k8sVersionStr string) (bool, error) {
 	}
 	return constraint.Check(k8sVersion), nil
 }
+
+// failIfContainsBadLogMsg makes a test case to fail if any message from
+// given log messages contains an entry from badLogMessages (map key) AND
+// does not contain ignore messages (map value).
+func failIfContainsBadLogMsg(logs string) {
+	for _, msg := range strings.Split(logs, "\n") {
+		for fail, ignoreMessages := range badLogMessages {
+			if strings.Contains(msg, fail) {
+				ok := false
+				for _, ignore := range ignoreMessages {
+					if strings.Contains(msg, ignore) {
+						ok = true
+						break
+					}
+				}
+				if !ok {
+					fmt.Fprintf(CheckLogs, "⚠️  Found a %q in logs\n", fail)
+					ginkgoext.Fail(fmt.Sprintf("Found a %q in Cilium Logs", fail))
+				}
+			}
+		}
+	}
+}

